share/functions/help.fish

```diff
@@ -127,17 +132,24 @@ function help --description 'Show help for the fish shell'
         # Go to the web. Only include one dot in the version string
         set -l version_string (echo $version| cut -d . -f 1,2)
         set page_url https://fishshell.com/docs/$version_string/$fish_help_page
+        # We don't need a trampoline for a remote URL.
+        set need_trampoline
     end
 
-    # OS X /usr/bin/open swallows fragments (anchors), so use osascript
-    # Eval is just a cheesy way of removing the hash escaping
-    if test "$fish_browser" = osascript
-        set -l opencmd 'open location "'(eval echo $page_url)'"'
-        osascript -e 'try' -e $opencmd -e 'on error' -e $opencmd -e 'end try'
-        return
+    if set -q need_trampoline[1]
+        # If string replace doesn't replace anything, we don't actually need a
+        # trampoline (they're only needed if there's a fragment in the path)
+        if set -l clean_url (string replace '\\#' '#' $page_url)
+            # Write a temporary file that will redirect where we want.
+            set -q TMPDIR
+            or set -l TMPDIR /tmp
+            set -l tmpdir (mktemp -d $TMPDIR/help.XXXXXX)
+            set -l tmpname $tmpdir/help.html
+            echo '<meta http-equiv="refresh" content="0;URL=\''$clean_url'\'" />' > $tmpname
+            set page_url file://$tmpname
+        end
     end
```

Comment:
v

Reply:
That doesn't seem to work, since `open` returns before launching the browser.